packages/docusaurus-utils/src/index.ts

```diff
@@ -191,10 +193,15 @@ export function parse(
 } {
   const options: {} = {
     excerpt: (file: matter.GrayMatterFile<string>): void => {
-      file.excerpt = file.content
-        .trim()
-        .split('\n', 1)
-        .shift();
+      let fileContent = file.content.trimLeft();
+
+      if (RegExp(importRegexString).test(fileContent)) {
```

Comment:
For better performance, I first check to see if import declarations are at the very beginning of the file and only then remove them. Perhaps there is a better way (?)

Reply:
Ideally we remove on the AST layer and not use regex, which can be unreliable. There can also be dynamic imports and CommonJS requires.

Reply:
Okay, should I close this PR?

Reply:
Honestly, we do not operate MDX AST in metadata, so I have no idea how to do it as you suggest. This case (remove `import`s) covers the main use case, and therefore resolves the issue..

Reply:
By the way, we still have a similar issue with MD rendering, for example https://v2.docusaurus.io/docs/lifecycle-apis

![image](https://user-images.githubusercontent.com/4408379/76166701-99567400-6171-11ea-9f9a-6b9a493f2762.png)

I have a solution for this, but it is in the same spirit as in current PR, so you are unlikely to like it.